File: app/src/main/java/io/github/sithengineer/motoqueiro/hardware/capture/MiBandData.java
package io.github.sithengineer.motoqueiro.hardware.capture;

public class MiBandData {

  private int heartRate;
  private boolean buttonPress;

  public MiBandData(int heartRate) {
    this.buttonPress = false;
    this.heartRate = heartRate;
  }

  public MiBandData(boolean buttonPress) {
    this.buttonPress = buttonPress;
  }

  public MiBandData() {
  }

  public boolean isButtonPress() {
    return buttonPress;
  }

  /**
   * @return true if heart rate is over 25bpm
   */
  public boolean hasHeartRate() {
    return heartRate > 25;
  }

  public int getHeartRateBpm() {
    return heartRate;
  }

  @Override public String toString() {
    return "heartRate=" + heartRate + ", buttonPress=" + buttonPress;
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/hardware/capture/MiBandData.kt
package io.github.sithengineer.motoqueiro.hardware.capture

class MiBandData {

  var heartRateBpm: Int = 0
  var isButtonPress: Boolean = false
    private set

  constructor(heartRate: Int) {
    this.isButtonPress = false
    this.heartRateBpm = heartRate
  }

  constructor(buttonPress: Boolean) {
    this.isButtonPress = buttonPress
  }

  constructor()

  /**
   * @return true if heart rate is over 25bpm
   */
  fun hasHeartRate(): Boolean {
    return heartRateBpm > 25
  }

  override fun toString(): String {
    return "heartRate=$heartRateBpm, buttonPress=$isButtonPress"
  }
}
